pybamm/models/submodels/interface/kinetics/total_kinetics.py

```diff
@@ -0,0 +1,242 @@
+
+                if self.half_cell:
+                    j0 = pybamm.concatenation(zero_s, j0_p)
+                    j0_dim = pybamm.concatenation(zero_s, j0_p_dim)
+                else:
+                    j0_n = variables[
+                        f"Negative electrode {reaction_name}exchange current density"
+                    ]
+                    j0_n_dim = variables[
+                        f"Negative electrode {reaction_name}"
+                        "exchange current density [A.m-2]"
+                    ]
+                    j0 = pybamm.concatenation(j0_n, zero_s, j0_p)
+                    j0_dim = pybamm.concatenation(j0_n_dim, zero_s, j0_p_dim)
+                new_variables.update(
+                    {
+                        f"{reaction_name}interfacial ".capitalize()
+                        + "current density": j,
+                        f"{reaction_name}interfacial ".capitalize()
+                        + "current density [A.m-2]": j_dim,
+                        f"{reaction_name}exchange ".capitalize()
+                        + "current density": j0,
+                        f"{reaction_name}exchange ".capitalize()
+                        + "current density [A.m-2]": j0_dim,
+                    }
+                )
+
+            # Sum variables
+            a_j_p = new_variables[
+                f"Positive electrode {reaction_name}volumetric "
+                "interfacial current density"
+            ]
+
+            if self.chemistry == "lithium-ion":
+                # Both the main reaction current contribute to the electrolyte reaction
+                # current
+                s_n, s_p = 1, 1
+            elif self.chemistry == "lead-acid":
+                if reaction_name == "":  # main reaction
+                    s_n, s_p = self.param.n.s_plus_S, self.param.p.s_plus_S
+                elif reaction_name == "oxygen ":
+                    s_n, s_p = self.param.s_plus_Ox, self.param.s_plus_Ox
+            if self.half_cell:
+                a_j = pybamm.concatenation(zero_s, a_j_p)
+                s = pybamm.concatenation(
+                    zero_s,
+                    pybamm.FullBroadcast(
+                        s_p, "positive electrode", "current collector"
+                    ),
+                )
+            else:
+                a_j_n = new_variables[
+                    f"Negative electrode {reaction_name}volumetric "
+                    "interfacial current density"
+                ]
+                a_j = pybamm.concatenation(a_j_n, zero_s, a_j_p)
+                s = pybamm.concatenation(
+                    pybamm.FullBroadcast(
+                        s_n, "negative electrode", "current collector"
+                    ),
+                    zero_s,
+                    pybamm.FullBroadcast(
+                        s_p, "positive electrode", "current collector"
+                    ),
+                )
+
+            # Override print_name
+            j.print_name = "J"
```

Comment:
For the print names, do we need to add labels of which reaction they are?

Reply:
probably, but that is a different issue